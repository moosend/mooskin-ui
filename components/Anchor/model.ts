import { IAnchorBoxComponentProps } from '../Box/model';

export interface IAnchorComponentProps extends IAnchorBoxComponentProps {
	/** Provide to make the Anchor disabled */
	disabled?: boolean;
}
