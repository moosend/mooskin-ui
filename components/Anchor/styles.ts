import styled from 'styled-components';

// Models
import { IAnchorComponentProps } from './model';

// Components
import { Box } from '../Box/Box';

// "CSS" Variables
import variables from '../_utils/globals/variables';

export const StyledAnchor = styled(Box)<IAnchorComponentProps>`
	cursor: ${(props) => (props.disabled ? 'not-allowed' : 'pointer')};
	outline: 0;
	font-size: 12px;
	&& {
		text-decoration: underline;
	}
	&:hover {
		text-decoration: none;
	}
	opacity: ${(props) => (props.disabled ? 0.5 : 1)};
	color: ${(props) => props.palette?.fontColors.secondary2 || variables.fontColors.secondary2};
`;

StyledAnchor.displayName = 'StyledAnchor';
