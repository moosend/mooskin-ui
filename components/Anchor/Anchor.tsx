import * as React from 'react';

// Mooskin Context HoC that passes context to component props
import { withMooskinContext } from '../Styled/MooskinContextProvider';

// Models
import { IAnchorComponentProps } from './model';

// Styled Components
import { StyledAnchor } from './styles';

/**
 * Anchor
 */
export const Anchor: React.FC<IAnchorComponentProps> = withMooskinContext((props) => {
	const onClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
		props.disabled && e.preventDefault();
		!props.disabled && props.onClick && props.onClick(e);
	};
	return <StyledAnchor {...props} onClick={onClick} boxAs="a" />;
});

Anchor.defaultProps = {
	className: '',
	style: {},
	target: '_blank',
};

Anchor.displayName = 'Anchor';
