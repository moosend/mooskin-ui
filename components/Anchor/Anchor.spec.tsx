import * as React from 'react';
import { Anchor } from './Anchor';

import { mount } from 'enzyme';

describe('Anchor', () => {
	test('renders correctly', () => {
		const func = jest.fn();

		const tree = mount(
			<Anchor onClick={func} disabled className="myClass" style={{ color: 'blue' }} id={'Anchor1'} href={'www.moosend.com'}>
				Mooskin
			</Anchor>
		);
		expect(tree).toMatchSnapshot();
	});

	test('renders properly into dom with color and label', () => {
		const func = jest.fn();

		const component = mount(
			<Anchor href="https://www.moosend.com" onClick={func}>
				asd
			</Anchor>
		);

		expect(component.find('StyledAnchor').text()).toBe('asd');
	});

	test('renders a an Anchor with href', () => {
		const func = jest.fn();

		const component = mount(
			<Anchor onClick={func} href={'https://www.moosend.com'}>
				asd
			</Anchor>
		);

		expect(component.find('StyledAnchor').prop('href')).toBe('https://www.moosend.com');
	});

	test('onClick prop callback is called when clicked', () => {
		const func = jest.fn();

		const component = mount(
			<Anchor href="https://www.moosend.com" onClick={func}>
				asd
			</Anchor>
		);
		component.find('StyledAnchor').simulate('click');
		expect(func).toHaveBeenCalled();
	});
});
