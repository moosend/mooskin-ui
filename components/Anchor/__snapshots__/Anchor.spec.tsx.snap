// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor renders correctly 1`] = `
<Anchor
  className="myClass"
  disabled={true}
  href="www.moosend.com"
  id="Anchor1"
  onClick={[MockFunction]}
  style={
    Object {
      "color": "blue",
    }
  }
  target="_blank"
>
  <Component
    className="myClass"
    disabled={true}
    href="www.moosend.com"
    id="Anchor1"
    onClick={[MockFunction]}
    palette={
      Object {
        "backgroundColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "borderColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
        "fontColors": Object {
          "background": "#FAFAFA",
          "black": "#000000",
          "darkgray1": "#777777",
          "darkgray2": "#454545",
          "gray1": "#E0E0E0",
          "gray2": "#E7E7E7",
          "medgray1": "#757575",
          "medgray2": "#888888",
          "primary1": "#4169E1",
          "primary2": "#212121",
          "primary3": "#00796B",
          "primary4": "#D32F2F",
          "secondary1": "#2E7D32",
          "secondary2": "#757575",
          "secondary3": "#673AB7",
          "secondary4": "#9C27B0",
          "secondary5": "#00796B",
          "secondary6": "#ECF0FC",
          "text": "#212121",
          "white": "#FFFFFF",
        },
      }
    }
    style={
      Object {
        "color": "blue",
      }
    }
    target="_blank"
  >
    <StyledAnchor
      boxAs="a"
      className="myClass"
      disabled={true}
      href="www.moosend.com"
      id="Anchor1"
      onClick={[Function]}
      palette={
        Object {
          "backgroundColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "borderColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
          "fontColors": Object {
            "background": "#FAFAFA",
            "black": "#000000",
            "darkgray1": "#777777",
            "darkgray2": "#454545",
            "gray1": "#E0E0E0",
            "gray2": "#E7E7E7",
            "medgray1": "#757575",
            "medgray2": "#888888",
            "primary1": "#4169E1",
            "primary2": "#212121",
            "primary3": "#00796B",
            "primary4": "#D32F2F",
            "secondary1": "#2E7D32",
            "secondary2": "#757575",
            "secondary3": "#673AB7",
            "secondary4": "#9C27B0",
            "secondary5": "#00796B",
            "secondary6": "#ECF0FC",
            "text": "#212121",
            "white": "#FFFFFF",
          },
        }
      }
      style={
        Object {
          "color": "blue",
        }
      }
      target="_blank"
    >
      <Box
        boxAs="a"
        className="sc-gtssRu AVBRk myClass"
        disabled={true}
        href="www.moosend.com"
        id="Anchor1"
        onClick={[Function]}
        palette={
          Object {
            "backgroundColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "borderColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
            "fontColors": Object {
              "background": "#FAFAFA",
              "black": "#000000",
              "darkgray1": "#777777",
              "darkgray2": "#454545",
              "gray1": "#E0E0E0",
              "gray2": "#E7E7E7",
              "medgray1": "#757575",
              "medgray2": "#888888",
              "primary1": "#4169E1",
              "primary2": "#212121",
              "primary3": "#00796B",
              "primary4": "#D32F2F",
              "secondary1": "#2E7D32",
              "secondary2": "#757575",
              "secondary3": "#673AB7",
              "secondary4": "#9C27B0",
              "secondary5": "#00796B",
              "secondary6": "#ECF0FC",
              "text": "#212121",
              "white": "#FFFFFF",
            },
          }
        }
        style={
          Object {
            "color": "blue",
          }
        }
        target="_blank"
        transition="0.2s all"
      >
        <Component
          boxAs="a"
          className="sc-gtssRu AVBRk myClass"
          disabled={true}
          href="www.moosend.com"
          id="Anchor1"
          onClick={[Function]}
          palette={
            Object {
              "backgroundColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "borderColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
              "fontColors": Object {
                "background": "#FAFAFA",
                "black": "#000000",
                "darkgray1": "#777777",
                "darkgray2": "#454545",
                "gray1": "#E0E0E0",
                "gray2": "#E7E7E7",
                "medgray1": "#757575",
                "medgray2": "#888888",
                "primary1": "#4169E1",
                "primary2": "#212121",
                "primary3": "#00796B",
                "primary4": "#D32F2F",
                "secondary1": "#2E7D32",
                "secondary2": "#757575",
                "secondary3": "#673AB7",
                "secondary4": "#9C27B0",
                "secondary5": "#00796B",
                "secondary6": "#ECF0FC",
                "text": "#212121",
                "white": "#FFFFFF",
              },
            }
          }
          style={
            Object {
              "color": "blue",
            }
          }
          target="_blank"
          transition="0.2s all"
        >
          <StyledBox
            as="a"
            boxAs="a"
            className="sc-gtssRu AVBRk myClass"
            disabled={true}
            href="www.moosend.com"
            id="Anchor1"
            onClick={[Function]}
            palette={
              Object {
                "backgroundColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "borderColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
                "fontColors": Object {
                  "background": "#FAFAFA",
                  "black": "#000000",
                  "darkgray1": "#777777",
                  "darkgray2": "#454545",
                  "gray1": "#E0E0E0",
                  "gray2": "#E7E7E7",
                  "medgray1": "#757575",
                  "medgray2": "#888888",
                  "primary1": "#4169E1",
                  "primary2": "#212121",
                  "primary3": "#00796B",
                  "primary4": "#D32F2F",
                  "secondary1": "#2E7D32",
                  "secondary2": "#757575",
                  "secondary3": "#673AB7",
                  "secondary4": "#9C27B0",
                  "secondary5": "#00796B",
                  "secondary6": "#ECF0FC",
                  "text": "#212121",
                  "white": "#FFFFFF",
                },
              }
            }
            style={
              Object {
                "color": "blue",
              }
            }
            target="_blank"
            transition="0.2s all"
          >
            <a
              className="sc-bdnylx jrLcJL sc-gtssRu AVBRk myClass"
              disabled={true}
              href="www.moosend.com"
              id="Anchor1"
              onClick={[Function]}
              style={
                Object {
                  "color": "blue",
                }
              }
              target="_blank"
            >
              Mooskin
            </a>
          </StyledBox>
        </Component>
      </Box>
    </StyledAnchor>
  </Component>
</Anchor>
`;
